### PR TITLE
ZCS-3870:GAL support for profile image

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Added GAL support for profile image
The thumbnail version profile image will be updated into LDAP into attribute 'thumbnailPhoto' upon SaveProfileImage Request
The thumbnail size will be 50*50 px(configurable in debug config)
The actual image will be stored into user database which can be accessible via rest URL
If the database update fails, the image will be removed from LDAP
The image base64 will be returned in SyncGalRequest/SearchGalRequest

ant test-all passed

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java [hunk 1]
```diff
@@ -73,41 +81,14 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
-                        && customData.get("p").equals("1")) {
-                        mbox.delete(octxt, item.getId(), MailItem.Type.DOCUMENT);
-                    }
-                }
-                imgDocItem = createDocument(imageDoc, zsc, octxt, mbox, null, in, folderId,
-                    MailItem.Type.DOCUMENT, null,
-                    new CustomMetadata(IMAGE_CUSTOM_DATA_SECTION, "d1:p1:1e"), false);
-            } catch (ServiceException e) {
-                throw ServiceException.FAILURE(FAILURE_MESSAGE, e);
+            if (inputSize > 2097152l) {
```
Comment (Prashantsurana) [inline store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java]: Can you please add 2097152l in debug config? 
2097152l seems bit odd number.  
can we have something more intuitive like 2Mb.

File: store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java [hunk 2]
```diff
@@ -118,9 +99,71 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                     ZimbraLog.mailbox.error("Exception in closing input stream for upload", e);
                 }
             }
+            if (!updateDBSuccess) {
+                // Databse update failed; revert from LDAP
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java]: updateDBSuccess will be false even if ldap update fails, in that case there is no need to revert from ldap.

File: store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java (hunk 1 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java]: Done. changed it to 2*1024*1024.

File: store/src/java/com/zimbra/cs/service/account/GetInfo.java [hunk 3]
```diff
@@ -275,6 +275,9 @@ private static int getProfileId(Mailbox mbox, OperationContext octxt) throws Ser
                 }
             }
         } catch (ServiceException exception) {
+            if (MailServiceException.NO_SUCH_FOLDER.equals(exception.getCode())) {
+                ZimbraLog.account.debug("Profile image folder doesn't exist");
+            }
             ZimbraLog.account.error("can't get profile image id", exception);
```
[the remark this answers is not in the document]
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/account/GetInfo.java]: Done

File: store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java (hunk 2 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java]: Done.

File: store/src/java/com/zimbra/cs/service/account/GetInfo.java [hunk 4]
```diff
@@ -275,7 +275,11 @@ private static int getProfileId(Mailbox mbox, OperationContext octxt) throws Ser
                 }
             }
         } catch (ServiceException exception) {
-            ZimbraLog.account.error("can't get profile image id", exception);
+            if (MailServiceException.NO_SUCH_FOLDER.equals(exception.getCode())) {
+                ZimbraLog.account.debug("Profile image folder doesn't exist");
+            } else {
+                ZimbraLog.account.error("can't get profile image id", exception);
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/account/GetInfo.java]: Split this 
ZimbraLog.account.error("can't get profile image id:%s", exception.getmessage());
ZimbraLog.account.debug("can't get profile image id", exception.getmessage());

**Pull request closed without merging**